dcpy/test/utils/test_s3.py

```diff
@@ -8,9 +9,27 @@
 TEST_FILE_NAME = "test-file"
 
 
+# assert that the default mock credentials work
+def test_client_create(aws_credentials):
+    s3.client()
+
+
+# assert client creation results for invalid client endpoint_url options
+def test_client_create_errors(aws_credentials):
+    with pytest.raises(TypeError, match="str expected, not NoneType"):
```

Comment:
wait, does this block actually raise an exception?

Reply:
the first `with pytest.raises` block? yea we can't set envars to non-strings

Reply:
Missing an `s3.client()` on line 21?

Reply:
Though having no endpoint is fine by boto3 standards - defaults to standard aws endpoint (we use DO hence our custom one). And something around the default endpoint behavior is how moto mocks it I think (hence the os.environ.pop in conf_test)

Reply:
Didn't see Damon's response, that makes sense. Is that worth testing in this case then? If its simply os.environ functionality, not related to client behavior

Reply:
see [a9a642f](https://github.com/NYCPlanning/data-engineering/pull/510/commits/a9a642feba597a0fdd1b57537a60c65b8aca37a9)

Reply:
That makes sense, let's leave it. Or maybe add a comment as to why it's there?